crates/vault-registry/src/types.rs

```diff
@@ -441,40 +441,59 @@ impl<T: Config> RichVault<T> {
         // Note: slashing of collateral must be called where this function is called (e.g. in Redeem)
     }
 
+    pub(crate) fn slash_collateral_and_transfer(&mut self, amount: Collateral<T>) -> DispatchResult {
```

Comment:
At first glance from a call to this function it wasn't clear where it was being transferred to. Maybe we can name it something like `slash_to_liquidation_vault`?